libnetdata/dyn_conf/dyn_conf.c

```diff
@@ -447,62 +526,104 @@ int register_module(struct configurable_plugin *plugin, struct module *module)
-                dictionary_set(module->jobs, job->name, job, sizeof(job));
-
-                deferred_config_push_back(plugin->name, module->name, job->name);
+        if (localhost) {
+            // load all jobs from disk
+            BUFFER *path = buffer_create(DYN_CONF_PATH_MAX, NULL);
+            buffer_sprintf(path, "%s/%s/%s", DYN_CONF_DIR, plugin->name, module->name);
+            DIR *dir = opendir(buffer_tostring(path));
+            if (dir != NULL) {
+                struct dirent *ent;
+                while ((ent = readdir(dir)) != NULL) {
+                    if (ent->d_name[0] == '.')
+                        continue;
+                    if (ent->d_type != DT_REG)
+                        continue;
+                    size_t len = strnlen(ent->d_name, NAME_MAX);
+                    if (len <= strlen(DYN_CONF_CFG_EXT))
+                        continue;
+                    if (strcmp(ent->d_name + len - strlen(DYN_CONF_CFG_EXT), DYN_CONF_CFG_EXT) != 0)
+                        continue;
+                    ent->d_name[len - strlen(DYN_CONF_CFG_EXT)] = '\0';
+
+                    struct job *job = job_new(ent->d_name);
+                    job->module = module;
+                    job->flags = JOB_FLG_PS_LOADED;
+                    job->type = JOB_TYPE_USER;
+
+                    dictionary_set(module->jobs, job->name, job, sizeof(job));
+
+                    deferred_config_push_back(plugins_dict, plugin->name, module->name, ent->d_name);
+                }
+                closedir(dir);
             }
-            closedir(dir);
+            buffer_free(path);
         }
-        buffer_free(path);
     }
 
     dictionary_set(plugin->modules, module->name, module, sizeof(module));
 
     return 0;
 }
 
+int register_job(DICTIONARY *plugins_dict, const char *plugin_name, const char *module_name, const char *job_name, enum job_type job_type, dyncfg_job_flg_t flags, int ignore_existing)
+{
+    int rc = 1;
+    const DICTIONARY_ITEM *plugin_item = dictionary_get_and_acquire_item(plugins_dict, plugin_name);
+    if (plugin_item == NULL) {
+        error_report("plugin \"%s\" not registered", plugin_name);
+        return 1;
+    }
+    struct configurable_plugin *plugin = dictionary_acquired_item_value(plugin_item);
+    struct module *mod = get_module_by_name(plugin, module_name);
+    if (mod == NULL) {
+        error_report("module \"%s\" not registered", module_name);
+        goto ERR_EXIT;
+    }
+    if (mod->type != MOD_TYPE_ARRAY) {
+        error_report("module \"%s\" is not an array", module_name);
+        goto ERR_EXIT;
+    }
+    if (get_job_by_name(mod, job_name) != NULL) {
+        if (!ignore_existing)
+            error_report("job \"%s\" already registered", job_name);
+        goto ERR_EXIT;
+    }
+
+    struct job *job = job_new(job_name);
+    job->module = mod;
+    job->flags = flags;
+    job->type = job_type;
+
+    dictionary_set(mod->jobs, job->name, job, sizeof(job));
```

Comment:
From coverity:

```
** CID 465042:  Incorrect expression  (SIZEOF_MISMATCH)
/libnetdata/dyn_conf/dyn_conf.c: 606 in register_job()
600     
601         struct job *job = job_new(job_name);
602         job->module = mod;
603         job->flags = flags;
604         job->type = job_type;
605     
>>>     CID 465042:  Incorrect expression  (SIZEOF_MISMATCH)
>>>     Passing argument "job" of type "struct job *" and argument "8UL /* sizeof (job) */" to function "dictionary_set_advanced" is suspicious.
606         dictionary_set(mod->jobs, job->name, job, sizeof(job));
607     
608         rc = 0;
609     ERR_EXIT:
610         dictionary_acquired_item_release(plugins_dict, plugin_item);
611         return rc;
```

Reply:
i miss something here? we are storing the pointer only not the struct?

Reply:
I also understood this as a false positive, but I worked few times with our dictionary library. @stelfrag can you confirm our understanding?

Reply:
So it just depends what you want to store and access it accordingly. So it is false suspicious

Reply:
intention here is to store pointer into dictionary, that is what the code does. Not sure if coverity will be happier if we use `sizeof(struct job *)` ?